src/control/crawler.c

```diff
@@ -840,6 +841,225 @@ void dt_control_crawler_show_image_list(GList *images)
+  sqlite3_stmt *stmt;
+
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                              "UPDATE main.images"
+                              " SET thumb_maxmip = ?2"
+                              " WHERE id = ?1",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 2, max_mip);
+  sqlite3_step(stmt);
+  sqlite3_finalize(stmt);
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                              "UPDATE main.images"
+                              " SET thumb_timestamp = ?2"
+                              " WHERE id = ?1",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+  DT_DEBUG_SQLITE3_BIND_INT64(stmt, 2, stamp);
+  sqlite3_step(stmt);
+  sqlite3_finalize(stmt);
+
+  dt_mimap_cache_evict(darktable.mipmap_cache, imgid);
+
+  // we have this in darktable-generate-cache but is it really good/necessary?
+  // dt_history_hash_set_mipmap(imgid);
+}
+
+static int _update_all_thumbs(const dt_mipmap_size_t max_mip)
+{
+  int missed = 0;
+  int updated = 0;
+  sqlite3_stmt *stmt;
+
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                              "SELECT id, import_timestamp, change_timestamp"
+                              " FROM main.images"
+                              " WHERE (thumb_timestamp < import_timestamp OR thumb_timestamp != change_timestamp OR thumb_maxmip < ?1)",
+                                -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, max_mip);
+  while(sqlite3_step(stmt) == SQLITE_ROW && _still_thumbing())
+  {
+    const dt_imgid_t imgid = sqlite3_column_int(stmt, 0);
+    const int64_t time_import = sqlite3_column_int64(stmt, 1);
+    const int64_t time_change = sqlite3_column_int64(stmt, 2);
+
+    char path[PATH_MAX] = { 0 };
+    gboolean from_cache = FALSE;
+    dt_image_full_path(imgid, path, sizeof(path), &from_cache);
+    const gboolean available = dt_util_test_image_file(path);
+
+    if(available)
+    {
+      _update_img_thumbs(imgid, max_mip, MAX(time_import, time_change));
+      updated++;
+    }
+    else
+      missed++;
+
+    dt_print(DT_DEBUG_CACHE,
+      "[thumb crawler] '%s' id=%d %savailable\n",
+      path, imgid, available ? "" : "NOT ");
+  }
+  sqlite3_finalize(stmt);
+
+  dt_print(DT_DEBUG_CACHE,
+    "[thumb crawler] max_mip=%d, %d thumbs updated, %d not found, %s.\n",
+      max_mip, updated, missed,
+      _still_thumbing() ? "all done" : "interrupted by user activity");
+
+  return updated;
+}
+
+static void _reinitialize_thumbs_database(const gboolean never)
+{
+  if(never)
+    dt_conf_set_string("backthumbs_mipsize", "never");
+
+  dt_print(DT_DEBUG_CACHE, "[thumb crawler] initialize database%s\n",
+    never ? ", switched to 'never'" : "");
+
+  sqlite3_stmt *stmt;
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
```

Comment:
You can merge the two SQL commands as with UPDATE you can set multiple column.

One question, why one would want to reinitialize the DB?

Reply:
Ok, if I read the code correctly when you select `reinitialize database` it switch to `never` after. I'm not sure this is good for user experience. Maybe we should add a new pref just to reset the database and always disabled the check just after having reset the DB. Thoughts?

Reply:
done